src/components/PopupWithForm/PopupWithForm.js

```diff
@@ -3,10 +3,12 @@ import "./PopupWithForm.css";
 import closeButton from "../../images/CloseButton.svg";
 const ModalWithForm = ({
   children,
+  error,
   onSignUpModal,
   onSignInModal,
   buttonText,
   title,
+  isValid,
   onCloseModal,
   onSubmit,
 }) => {
```

Comment:
**Could be improved**

If it’s interesting for you here is a way to make a special wrapper for any `modal` with cross button, `Escape` and `overlay` listeners inside. You should create component `Modal`, store it in `Modal.js`. 


```jsx

import { useEffect } from "react";

export const Modal = ({ name, onClose, children }) => {
  // here is `useEffect` for the `Escape` listener
  useEffect(() => {
    // we should define the handler inside `useEffect`, so that it wouldn’t lose the reference to be able to remove it
    const handleEscape = (e) => {
      if (e.key === "Escape") {
        onClose();
      }
    };

    document.addEventListener("keydown", handleEscape);
    // don’t forget to remove the listener in the `clean-up` function
    return () => document.removeEventListener("keydown", handleEscape);
  }, [onClose]);

  // here is the overlay handler
  const handleOverlay = (e) => {
    if (e.target === e.currentTarget) {
      onClose();
    }
  };

  // then we add the main wrapper with class `modal`
  return (
    <div className={`modal modal_type_${name}`} onClick={handleOverlay}>
      {/* the container for the contents */}
      <div className="modal__container">
        {/* here will be anything you add as `children`*/}
        {children}
        {/* add the close button */}
        <button className="modal__close" type="button" onClick={onClose} />
      </div>
    </div>
  );
};



```


And now you can use `Modal` for any popup in the project:  `ItemModal`  and `ModalWithForm` 

```jsx


function ModalWithForm({ name, onClose, ...props}) {
  return (
    <Modal name={name} onClose={onClose}>
        <h2 className='popup__title'>{props.title}</h2>


```

You can make also `Input` for inputs and `Form` for any form in the project.

Reply:
Thank you Gennady for the feedback. I like the way you built a fuction for fetch it's clear and easy to use. Also the way of using the useEffect in the handler seemed interesting. I'm gonna use you advice in improving this project I agree spliting my components to smaller ones would make it smoother and easier to debug!